posthog/hogql/database/schema/util/where_clause_extractor.py

```diff
@@ -427,3 +552,31 @@ def visit_field(self, node: ast.Field) -> ast.Field:
 
     def visit_alias(self, node: ast.Alias) -> ast.Expr:
         return self.visit(node.expr)
+
+
+def flatten_ands(exprs):
+    flattened = []
+    for expr in exprs:
+        if isinstance(expr, ast.And):
+            flattened.extend(flatten_ands(expr.exprs))
+        else:
+            flattened.append(expr)
+    return flattened
+
+
+def flatten_ors(exprs):
+    flattened = []
+    for expr in exprs:
+        if isinstance(expr, ast.Or):
+            flattened.extend(flatten_ors(expr.exprs))
+        else:
+            flattened.append(expr)
+    return flattened
+
+
+def is_not_truthy(value):
+    return value is False or value is None or value == 0
```

Comment:
Do you want to check the string `"0"` too?

Reply:
I think not:
- Strings aren't truthy in SQL 😅. They're either null or they're a string.
- True/False is actually converted to 1 / 0 in ClickHouse, so it's the same thing
- Sometimes we pass in our own AST nodes with true/false, without converting them to the clickhouse-native 1/0.

Thus just in case I won't go for "javascript is magic" level string comparisons 🙈